jwst/model_blender/tablebuilder.py

```diff
@@ -0,0 +1,57 @@
+import numpy as np
+
+
+def _convert_dtype(value):
+    """Convert numarray column dtype into YAML-compatible format description"""
+    if 'U' in value:
+        # working with a string description
+        str_len = int(value[value.find('U') + 1:])
+        new_dtype = ['ascii', str_len]
+    elif value == 'bool':
+        # cast all bool to int8 to avoid issues on write
+        new_dtype = 'int8'
+    else:
+        new_dtype = str(value)
+
+    return new_dtype
+
+
+def table_to_schema(table):
+    return {
+        'title': 'Combined header table',
+        'fits_hdu': 'HDRTAB',
+        'datatype': [
+            {
+                'name': col_name,
+                'datatype': _convert_dtype(str(table.dtype[col_name])),
+            }
+            for col_name in table.dtype.fields
+        ],
+    }
+
+
+class TableBuilder:
```

Comment:
can you add short docstrings to this class and its functions?

Reply:
Good idea! I added docstrings in: https://github.com/spacetelescope/jwst/pull/8759/commits/68d9f6b149b3c1d4b57a330a524c19d831b10799
(and added some for `ModelBlender`).